### ISSUE TITLE
Security concern with activesupport gem

### ISSUE THREAD

**Issue by aaulthudl** [CocoaPods/CocoaPods#12040]
* [ 🌈] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

`activesupport` gem which is defined [here](https://github.com/CocoaPods/CocoaPods/blob/53893a2de0d09827fcdaaf089ac25a98d85ed074/Gemfile#L37) has a security concern.

Issue flagged by Snyk:
<img width="1353" alt="image" src="https://github.com/CocoaPods/CocoaPods/assets/84314259/3376e913-9a0d-4922-8510-96292dfc99a4">

Issue flagged by GitHub Dependabot:
<img width="1504" alt="image" src="https://github.com/CocoaPods/CocoaPods/assets/84314259/d09e609f-8945-4ed2-a642-8019f8aa9a28">

## What did you expect to happen?

No security issues.

## What happened instead?

Security issue flagged for `CocoaPods` due to `activesupport` dependency.

## CocoaPods Environment

N/A

## Project that demonstrates the issue

N/A

**Comment by orta** [comment]
CocoaPods Core's gemspec (which is the one including activerecord) has a version specifier which would include all the fixes: 

https://github.com/CocoaPods/Core/blob/9320bf557741d7ae80eda4415bc6fb2b0b3a8b9c/cocoapods-core.gemspec#L22C1-L22C60

```
  s.add_runtime_dependency 'activesupport', '>= 5.0', '< 8'
```

So there's no need for anything on our side, you can run `bundle update activerecord` on your own project to fix this 👍🏻

**Comment by aaulthudl** [comment]
Is there a reason this is added here instead of in the `Gemfile`? In there it says `gem 'activesupport', '> 5', '< 7'`

**Comment by orta** [comment]
A Gemfile = Podfile, a gemspec = podspec. 

The Gemfile in this repo doesn't doesn't affect users of CocoaPods

**Comment by aaulthudl** [comment]
TIL! Thanks for your help. Is there the conversation for pinning the `activesupport` version in the `gemspec` to be one which contains the fixed security concern?

**Comment by orta** [comment]
I don't think we need it, because we're not constraining anyone to use those specific versions. Our constraints say greater than 5, less than 8 which allows for all the fixed versions. 

That said, I imagine it can't hurt to bump the lower bound from `>= 5` to `>= 5.2`, I'd be open to that PR to core
